pkg/kubelet/stats/cri_stats_provider_windows_test.go

```diff
@@ -0,0 +1,424 @@
+					{
+						container: hcsshim.ContainerProperties{
+							ID: "c3",
+						}, hcsStats: []hcsshim.NetworkStats{
+							{
+								BytesReceived: 3,
+								BytesSent:     30,
+								EndpointId:    "test2",
+								InstanceId:    "test3",
+							},
+						},
+					},
+				},
+			},
+			want: map[string]*statsapi.NetworkStats{
+				"c1": {
+					Time: v1.NewTime(shimTime),
+					InterfaceStats: statsapi.InterfaceStats{
+						Name:    "test",
+						RxBytes: toP(1),
+						TxBytes: toP(10),
+					},
+					Interfaces: []statsapi.InterfaceStats{
+						{
+							Name:    "test",
+							RxBytes: toP(1),
+
+							TxBytes: toP(10),
+						},
+					},
+				},
+				"c2": {
+					Time: v1.Time{},
+					InterfaceStats: statsapi.InterfaceStats{
+						Name:    "test2",
+						RxBytes: toP(2),
+						TxBytes: toP(20),
+					},
+					Interfaces: []statsapi.InterfaceStats{
+						{
+							Name:    "test2",
+							RxBytes: toP(2),
+							TxBytes: toP(20),
+						},
+					},
+				},
+				"c3": {
+					Time: v1.Time{},
+					InterfaceStats: statsapi.InterfaceStats{
+						Name:    "test2",
+						RxBytes: toP(2),
+						TxBytes: toP(20),
+					},
+					Interfaces: []statsapi.InterfaceStats{
+						{
+							Name:    "test2",
+							RxBytes: toP(2),
+							TxBytes: toP(20),
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "multiple stats instances of same interface only picks up first",
+			fields: fakehcsshim{
+				containers: []containerStats{
+					{
+						container: hcsshim.ContainerProperties{
+							ID: "c1",
+						}, hcsStats: []hcsshim.NetworkStats{
+							{
+								BytesReceived: 1,
+								BytesSent:     10,
+								EndpointId:    "test",
+								InstanceId:    "test",
+							},
+							{
+								BytesReceived: 3,
+								BytesSent:     30,
+								EndpointId:    "test",
+								InstanceId:    "test3",
+							},
+						},
+					},
+					{
+						container: hcsshim.ContainerProperties{
+							ID: "c2",
+						}, hcsStats: []hcsshim.NetworkStats{
+							{
+								BytesReceived: 2,
+								BytesSent:     20,
+								EndpointId:    "test2",
+								InstanceId:    "test2",
+							},
+						},
+					},
+				},
+			},
+			want: map[string]*statsapi.NetworkStats{
+				"c1": {
+					Time: v1.NewTime(shimTime),
+					InterfaceStats: statsapi.InterfaceStats{
+						Name:    "test",
+						RxBytes: toP(1),
+						TxBytes: toP(10),
+					},
+					Interfaces: []statsapi.InterfaceStats{
+						{
+							Name:    "test",
+							RxBytes: toP(1),
+
+							TxBytes: toP(10),
+						},
+					},
+				},
+				"c2": {
+					Time: v1.Time{},
+					InterfaceStats: statsapi.InterfaceStats{
+						Name:    "test2",
+						RxBytes: toP(2),
+						TxBytes: toP(20),
+					},
+					Interfaces: []statsapi.InterfaceStats{
+						{
+							Name:    "test2",
+							RxBytes: toP(2),
+							TxBytes: toP(20),
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "multiple endpoints per container",
```

Comment:
Ignorance incoming: Are there Windows CNI's that allocate multiple endpoints per pod?

Reply:
I am not aware of any but the previous implementation did handle the scenario which is why added this test case

Reply:
Gotcha, feel free to resolve